plugins/inputs/etherscan/balance.go

```diff
@@ -0,0 +1,282 @@
+package etherscan
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/big"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	ErrBalanceAddressCountExhausted = errors.New("maximum address limit reached for batch request")
+	ErrReturnCodeNotOK              = errors.New("return code was not OK")
+	ErrNotValidEthAddress           = errors.New("invalid Ethereum address")
+)
+
+type ValidatingBalanceResponder interface {
+	ValidateResponseStatus() error
+	Result() []AccountBalance
+}
+
+type balance struct {
+	Network string `toml:"network"`
+	Address string `toml:"address"`
+	Tag     string `toml:"tag"`
+}
+
+type MultiBalanceResponse struct {
+	Status  string           `json:"status"`
+	Message string           `json:"message"`
+	Results []AccountBalance `json:"result"`
+}
+
+func (mbr *MultiBalanceResponse) ValidateResponseStatus() error {
+	status, err := strconv.ParseInt(mbr.Status, 10, 64)
+	if err != nil {
+		return fmt.Errorf("error processing response status [%w]", err)
+	}
+
+	if err := validateReturnCode(status); err != nil {
+		if mbr.Message == "OK" {
+			return nil
+		}
+
+		return fmt.Errorf("%s error: [%s]", err.Error(), mbr.Message)
+	}
+
+	return nil
+}
+
+func (mbr *MultiBalanceResponse) Result() []AccountBalance {
+	return mbr.Results
+}
+
+type SingleBalanceResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message"`
+	Results string `json:"result"`
+}
+
+func (sbr *SingleBalanceResponse) ValidateResponseStatus() error {
+	status, err := strconv.ParseInt(sbr.Status, 10, 64)
+	if err != nil {
+		return fmt.Errorf("error processing response status [%w]", err)
+	}
+
+	if err := validateReturnCode(status); err != nil {
+		if sbr.Message == "OK" {
+			return nil
+		}
+
+		return fmt.Errorf("%s error: [%s]", err.Error(), sbr.Message)
+	}
+
+	return nil
+}
+
+func (sbr *SingleBalanceResponse) Result() []AccountBalance {
+	return []AccountBalance{
+		{
+			Account: "",
+			Balance: sbr.Results,
+		},
+	}
+}
```

Comment:
Can you please get rid of this and change `Send()`, where you know if you will receive a single or multiple responses, to abstract the type!? This will save quite come code...